### ISSUE TITLE
Can't install pre-1.6.x over pre-1.6.x

### ISSUE THREAD

**Issue by esnible** [istio/istio#22676]
I installed using istioctl from HEAD on March 30th and again on April 2

```
istioctl install \
   --set hub=gcr.io/istio-testing \
   --set tag=latest \
   --set values.global.mtls.enabled=false \
   --set values.global.imagePullPolicy=Always \
   --set revision=revtest
...
The Deployment "istiod-revtest" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"istiod", "istio.io/rev":"revtest"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable (repeated 1 times)
...
```

This shouldn't be a problem for users of releases.  It shows a flaw in the `manifest apply` logic; which perhaps should delete and re-apply under some conditions.

**Comment by howardjohn** [comment]
We do not support (in place) upgrades between prereleases, only officially published versions.

I don't think deleting and reapplying is a good idea, we used to do that. If the user wants to they can pass `--force` (I think - this might not be true) or manually delete it. Deleting ourselves is a risky move, and due to a bug(?) in k8s it doesn't work: https://github.com/kubernetes/kubernetes/issues/89322

**Comment by esnible** [comment]
I am not expected support!  I am happy to close this issue.  I created this issue because I suspect without a fix it will not be possible to upgrade from 1.5.x to 1.6.x.

`--force` doesn't work because it is the K8s API that rejects the application, not the operator code.

**Comment by howardjohn** [comment]
There is two layers of --force. If you pass --force to kubectl it will do what you said (delete and re-apply), but I don't know if we pass through our --force to kubectl --force.

There is a lot of risk about this type of change, see https://github.com/istio/istio/issues/22680 for example. I just added tests that should end these issues for good, since right now I keep fixing one thing and breaking another https://github.com/istio/istio/pull/22681.

**Comment by howardjohn** [comment]
We fixed this issue in the latest 1.6 so 1.5 -> 1.6 upgrade is working. We also have tests to prevent regression
